app/lib/actions/katello/content_view/promote_to_environment.rb

```diff
@@ -41,7 +41,7 @@ def humanized_name
 
         def run
           environment = ::Katello::KTEnvironment.find(input[:environment_id])
-          if ::Katello::CapsuleContent.sync_needed?(environment)
+          if ::Katello::CapsuleContent.sync_needed?(environment) && Setting[:foreman_proxy_content_auto_sync] == true
```

Comment:
I have to look further, but my first thought is why not gate this at the step that actually calls `PromoteToEnvironment` rather than in the action itself?

Reply:
> Should  this all be happening in plan phase instead of run? it seems a little strange to be calling async_task from inside run phase of another task, vs laying out all tasks in plan and sending a param to CapsuleSync task to not raise a failure when a sync fails. This change would be outside the scope of this PR though and would need a new PR.

The reason we do this is that we don't want the publish/promote to wait on the capsule syncs.  It could take hours and hours for a capsule to sync a new content view, but the publish and promote is actually already done.